aea/aea_builder.py

```diff
@@ -1196,30 +1227,26 @@ def _load_and_add_skills(self, context: AgentContext) -> None:
                 skill.skill_context.set_agent_context(context)
                 skill.skill_context.logger = logging.getLogger(logger_name)
             else:
+                configuration = deepcopy(configuration)
                 skill_context = SkillContext()
                 skill_context.set_agent_context(context)
                 skill_context.logger = logging.getLogger(logger_name)
-                try:
-                    skill = Skill.from_config(
-                        configuration, skill_context=skill_context
-                    )
-                except ModuleNotFoundError as e:
-                    _handle_error_while_loading_component_module_not_found(
-                        configuration, e
-                    )
-                except Exception as e:
-                    _handle_error_while_loading_component_generic_error(
-                        configuration, e
-                    )
-            self._add_component_to_resources(skill)
+                skill = cast(
+                    Skill,
+                    load_component_from_config(
+                        ComponentType.SKILL, configuration, skill_context=skill_context
+                    ),
+                )
+            resources.add_component(skill)
 
     def _load_connection(
-        self, address: Address, configuration: ConnectionConfig
+        self, identity: Identity, wallet: Wallet, configuration: ConnectionConfig
```

Comment:
every connection will have access to the wallet?

Reply:
No; only the Connection `CryptoStore`